doc: default priority class (#808)
longhorn/website: content/docs/1.6.0/deploy/important-notes/index.md

reattached (by scaling its consuming workload up). Consider scaling workloads down and back up again as soon as possible
after upgrading from a version without this mechanism (v1.5.1 or older) to v{{< current-version >}}.

### Default Priority Class

> Longhorn v1.6.0 introduces the default priority class `longhorn-critical` with the highest value to ensure longhorn pods will not be evicted when low system resources by the Kube scheduler.

Author: Modified. Thanks.

> Longhorn v1.6.0 introduces the default priority class `longhorn-critical` with the highest value to ensure longhorn pods have not been evicted when low system resources by the Kube scheduler.

Reviewer: "... will not be evicted ..."

Author: Modified. Thanks

> For upgrade:

Reviewer 2: ```suggestion
During upgrades, Longhorn applies the default Priority Class to components depending on specific settings.
```

Author: Modified. Thanks

> - When all volumes are detached and the Priority Class default setting is not customized, the default Priority Class is applied to all Longhorn components. The default Priority Class setting `priority-class` is updated.

Reviewer 2: ```suggestion
- When all volumes are detached and you did not specify a value for the global Priority Class setting `priority-class`, the default Priority Class is applied to all Longhorn components. `priority-class` is updated.
```

Author: Modified. Thanks

> - When all volumes are detached and you have specified a value for the Priority Class default setting, the default Priority Class is applied only to user-deployed components. The default Priority Class setting `priority-class` is not updated.

Reviewer 2: > The default Priority Class setting `priority-class` is not updated.

The meaning of sentence 1 has now changed so I need to clarify the meaning of sentence 2. Do you mean this:
"The default Priority Class setting `priority-class` **of system-managed components** is not updated."

Author: > The default Priority Class setting priority-class is not updated

That's a global setting `priority-class` https://longhorn.io/docs/1.5.3/references/settings/#priority-class and it will apply to all components if it is changed.

Maybe it should be 
> The global setting `priority-class` is not updated

> - When all volumes are detached and you have specified a value for the Priority Class default setting, the default Priority Class is applied only to user-deployed components. The default Priority Class setting `priority-class` is not updated.

Reviewer 2: ```suggestion
- When all volumes are detached and you specified a value for the global Priority Class setting `priority-class`, the default Priority Class is applied only to user-deployed components. `priority-class` is not updated.
```

Author: Modified. Thanks

Author: Modified. Thanks

> 1. If all volumes are detached, the setting will be applied including user-managed components. The setting of the priority class will be updated because the source of truth of settings is the config map (longhorn-default-setting).
> 2. If all volumes are detached and the setting of priority class is set to a priority class, the setting should be applied except for user-managed components. The setting of the priority class will not be updated.
> 3. If volumes are attached, the setting should be applied except for user-managed components. The setting of the priority class will not be updated.

Reviewer 2: ```suggestion
- When all volumes are detached and the Priority Class default setting is not customized, the default Priority Class is applied to all Longhorn components. 

- When all volumes are detached and you have specified a value for the Priority Class default setting, the default Priority Class is applied only to system-managed components. The Priority Class default setting for user-deployed components is not updated.

- When one or more volumes are attached, the default Priority Class is applied only to system-managed components. The Priority Class default setting for user-deployed components is not updated.
```

Reviewer 2: - This is not a procedure so we should use an unordered list.
- I rewrote the descriptions to clearly describe the conditions and affected components.

Author: Modified. Thanks

Author: And I modified the user-deployed components and system-managed components parts.
System-managed components would be affected only when the default Priority Class setting `priority-class` can be updated.

> - When one or more volumes are attached, the default Priority Class is applied only to user-deployed components only if they haven't been set. The default Priority Class setting `priority-class` is not updated.

Reviewer 2: > applied only to user-deployed components only if they haven't been set

What do you mean by "if they haven't been set"?

Author: >"if they haven't been set"

It means that if users do not set the `priorityClass` in user-managed components such as https://github.com/longhorn/longhorn/blob/master/chart/values.yaml#L351-L356, the default Priority Class `longhorn-critical` would be set to the `prioriClass`.

> - When one or more volumes are attached, the default Priority Class is applied only to user-deployed components only if they haven't been set. The default Priority Class setting `priority-class` is not updated.

Reviewer 2: ```suggestion
- When one or more volumes are attached and you did not specify a value for the global Priority Class setting `priority-class`, the default Priority Class is applied only to user-deployed components. `priority-class` is not updated.
```

Author: Modified. Thanks

> If you want to make it take effect for item 2,3, you need to detach all volumes and update the priority class on your own after the upgrade.
> 
> You can change these behaviors by modifying priority class parameters in the `value.yaml` of the Longhorn Helm chart or [longhorn/deploy/longhorn.yaml](https://raw.githubusercontent.com/longhorn/longhorn/v1.6.0/deploy/longhorn.yaml) before the upgrade.

Reviewer 2: ```suggestion
If you want to apply the default Priority Class to user-deployed components, you must detach all volumes and change the Priority Class default setting value after the upgrade is successfully completed.

You can change these behaviors by modifying the `priority-class` parameters in the [Longhorn Helm chart](https://github.com/longhorn/longhorn/blob/master/chart/) `values.yaml` file or the config map in [longhorn/deploy/longhorn.yaml](https://raw.githubusercontent.com/longhorn/longhorn/v1.6.0/deploy/longhorn.yaml) before starting the upgrade process.
```

Author: Modified. Thanks.

> You can change these behaviors by modifying the parameters `defaultSetting.priorityClass` or `priorityClass` of user-deployed components in the [Longhorn Helm chart](https://github.com/longhorn/longhorn/blob/master/chart/) `values.yaml` file or the config map in [longhorn/deploy/longhorn.yaml](https://raw.githubusercontent.com/longhorn/longhorn/v1.6.0/deploy/longhorn.yaml) before starting the upgrade process.

Reviewer 2: The sentence is now too long because of the added details. We can list the options to help users scan the info more quickly.

> You can change these behaviors by modifying the parameters `defaultSetting.priorityClass` or `priorityClass` of user-deployed components in the [Longhorn Helm chart](https://github.com/longhorn/longhorn/blob/master/chart/) `values.yaml` file or the config map in [longhorn/deploy/longhorn.yaml](https://raw.githubusercontent.com/longhorn/longhorn/v1.6.0/deploy/longhorn.yaml) before starting the upgrade process.

Author: That would be good, and I would modify it if you have any suggestions for the review conversion below.

> You can change these behaviors by modifying the parameters `defaultSetting.priorityClass` or `priorityClass` of user-deployed components in the [Longhorn Helm chart](https://github.com/longhorn/longhorn/blob/master/chart/) `values.yaml` file or the config map in [longhorn/deploy/longhorn.yaml](https://raw.githubusercontent.com/longhorn/longhorn/v1.6.0/deploy/longhorn.yaml) before starting the upgrade process.

Author: Modified. Thanks

During upgrades, Longhorn applies the default Priority Class to components depending on specific settings.

- When all volumes are detached and you did not specify a value for the global Priority Class setting `priority-class`, the default Priority Class is applied to all Longhorn components. `priority-class` is updated.
- When all volumes are detached and you specified a value for the global Priority Class setting `priority-class`, the default Priority Class is applied only to user-deployed components. `priority-class` is not updated.
- When one or more volumes are attached and you did not specify a value for `PriorityClass` in the `chart/value.yaml` or `longhorn/deploy/longhorn.yaml`, the default Priority Class is applied only to user-deployed components. `priority-class` is not updated.

If you want to apply the default Priority Class to system-managed components, you must detach all volumes and change the Priority Class default setting value after the upgrade is successfully completed.

> You can change these behaviors by customizing the following before starting the upgrade process:
> - For user-deployed components: `defaultSetting.priorityClass` or `priorityClass` parameters in the `values.yaml` file of the [Longhorn Helm chart](https://github.com/longhorn/longhorn/blob/master/chart/)
> - Config map in [longhorn/deploy/longhorn.yaml](https://raw.githubusercontent.com/longhorn/longhorn/v1.6.0/deploy/longhorn.yaml)
> 

Reviewer 3: ```suggestion
You can change these behaviors by customizing the following before starting the upgrade process:

- For user deployed components: `priorityClass` parameters for each component in the `values.yaml` file of the [Longhorn Helm chart](https://github.com/longhorn/longhorn/blob/v1.6.0/chart/values.yaml)
- For system managed components: `defaultSetting.priorityClass` in the `values.yaml` file of the [Longhorn Helm chart](https://github.com/longhorn/longhorn/blob/v1.6.0/chart/values.yaml)
```

For ConfigMap `longhorn-default-setting`, it's redundant, because it has been covered in `defaultSetting.priorityClass`, right?

Author: Yes. Just mentioned that deploying by the manifest YAML file.
Modified. Thanks.
